test/integration/controlplane/transformation/kms_transformation_test.go

```diff
@@ -536,6 +539,91 @@ resources:
 	}
 }
 
+func TestEncryptAll(t *testing.T) {
+	testCases := []struct {
+		name             string
+		encryptionConfig string
+	}{
+		{
+			name: "encrypt all resources",
+			encryptionConfig: `
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+  - resources:
+    - '*.*'
+    providers:
+    - kms:
+        name: encrypt-all-kms-provider
+        cachesize: 1000
+        endpoint: unix:///@encrypt-all-kms-provider.sock		
+`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pluginMock, err := mock.NewBase64Plugin("@encrypt-all-kms-provider.sock")
+			if err != nil {
+				t.Fatalf("failed to create mock of KMS Plugin: %v", err)
+			}
+
+			go pluginMock.Start()
+			if err := mock.WaitForBase64PluginToBeUp(pluginMock); err != nil {
+				t.Fatalf("Failed start plugin, err: %v", err)
+			}
+			defer pluginMock.CleanUp()
+
+			test, err := newTransformTest(t, tc.encryptionConfig, false, "")
+			if err != nil {
+				t.Fatalf("failed to start KUBE API Server with encryptionConfig")
+			}
+			defer test.cleanUp()
+
+			_, serverResources, err := test.restClient.Discovery().ServerGroupsAndResources()
```

Comment:
You need to enable all feature gates and all resources (see the etcd storage path test)